mev-boost-rs/src/relay_mux.rs

```diff
@@ -143,11 +145,40 @@ impl BlindedBlockProvider for RelayMux {
         &self,
         bid_request: &BidRequest,
     ) -> Result<SignedBuilderBid, BlindedBlockProviderError> {
-        let responses = stream::iter(self.relays.iter().cloned())
+        let time_out =
+            tokio::time::sleep(tokio::time::Duration::from_secs(FETCH_BEST_BID_TIME_OUT));
+        tokio::pin!(time_out);
+
+        let mut requests = stream::iter(self.relays.iter().cloned())
             .map(|relay| async move { relay.fetch_best_bid(bid_request).await })
-            .buffer_unordered(self.relays.len())
-            .collect::<Vec<_>>()
-            .await;
+            .buffer_unordered(self.relays.len());
+
+        let mut responses = Vec::new();
+
+        loop {
+            tokio::select! {
+                res = requests.next() => {
+                    match res {
+                        Some(bid) => {
+                            responses.push(bid);
+                        }
+                        None => {
+                            break;
+                        }
+                    }
+                }
+                _ = &mut time_out => {
+                    break;
+                }
+            }
+        }
+
+        tracing::debug!(
+            "{} out of {} relays fetched bids before {} seconds time out",
+            responses.len(),
+            self.relays.len(),
+            FETCH_BEST_BID_TIME_OUT
```

Comment:
this is a nice log -- I'd even consider comparing `responses.len()` to `self.relays.len()` and if they are equal, do nothing; if they are _unequal_, then log a `tracing::warning!`

that way laggy relays will be surfaced to the user